cmd/kube-rbac-proxy/app/setup_authorizer_test.go

```diff
@@ -0,0 +1,430 @@
+/*
+Copyright 2025 the kube-rbac-proxy maintainers. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/brancz/kube-rbac-proxy/pkg/authorization"
+	"github.com/brancz/kube-rbac-proxy/pkg/authorization/rewrite"
+	"github.com/brancz/kube-rbac-proxy/pkg/authorization/static"
+	"github.com/brancz/kube-rbac-proxy/pkg/server"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	serverconfig "k8s.io/apiserver/pkg/server"
+)
+
+type mockAuthorizer func(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error)
+
+func (m mockAuthorizer) Authorize(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error) {
+	return m(ctx, a)
+}
+
+func TestSetupAuthorizer_AllowPathsWithRewriteAndStaticAuth(t *testing.T) {
+	mockDelegated := mockAuthorizer(
+		func(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error) {
+			if a.GetUser().GetName() == "system:serviceaccount:default:client-with-rbac" {
+				return authorizer.DecisionAllow, "delegated allow", nil
+			}
+			return authorizer.DecisionNoOpinion, "i don't care", nil
+		},
+	)
+
+	mockErr := mockAuthorizer(
+		func(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error) {
+			return authorizer.DecisionDeny, "delegated deny", fmt.Errorf("delegated error")
+		},
+	)
```

Comment:
you could base the decision off of a username so that you don't have to have an extra argument to your tests